internal/locate/region_request.go

```diff
@@ -1431,6 +1437,12 @@ func (s *RegionRequestSender) SendReqCtx(
 			}
 		}
 
+		if e := tikvrpc.SetContext(req, rpcCtx.Meta, rpcCtx.Peer); e != nil {
+			return nil, nil, retryTimes, err
+		}
+		rpcCtx.contextPatcher.applyTo(&req.Context)
+		patchRequestSource(req, requestSource, readType, retryTimes)
```

Comment:
If finally "nil, nil" is returned, the next round `RegionRequestSender.SendReqCtx` retry, stale flag would not be considered again, in this case it seems the `readType` may not represent the original read type.

Reply:
I change the implemention and it now handles the retry from upper layer.